### PR TITLE
Added C++ implementation, load balancer support, and user restriction to Pregel API

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/504/

File: arangod/Pregel/PregelFeature.cpp [hunk 1]
```diff
@@ -87,29 +241,38 @@ void PregelFeature::beginShutdown() {
...
-void PregelFeature::addConductor(std::unique_ptr<Conductor>&& c, uint64_t executionNumber) {
+void PregelFeature::addConductor(std::unique_ptr<Conductor>&& c,
+                                 uint64_t executionNumber) {
   MUTEX_LOCKER(guard, _mutex);
-  //_executions.
-  _conductors.emplace(executionNumber, std::shared_ptr<Conductor>(c.get()));
+  std::string user = ExecContext::CURRENT ? ExecContext::CURRENT->user() : "";
+  _conductors.emplace(
+      executionNumber,
+      std::make_pair(user, std::shared_ptr<Conductor>(c.get())));
```
Comment (graetzer) [inline arangod/Pregel/PregelFeature.cpp]: couldn't you store the user on the conductor ?

File: js/actions/api-pregel.js [hunk 2]
```diff
@@ -62,18 +62,18 @@ function post_api_pregel (req, res) {
   if (!json.algorithm || typeof json.algorithm !== 'string') {
```
Comment (graetzer) [inline js/actions/api-pregel.js]: can't we delete this file now ?

Comment (danielhlarkin) [reply js/actions/api-pregel.js]: Yes, we can. I thought I had deleted it already, but clearly I was mistaken.

File: arangod/Pregel/PregelFeature.cpp (hunk 1 above)
Comment (danielhlarkin) [reply arangod/Pregel/PregelFeature.cpp]: You could, yes. It's fairly arbitrary which way to do it. This matches the way I implemented it in the other handlers, as there were some structures that I didn't necessarily want to change elsewhere.

**Pull request merged**